src/aspire/basis/polar_2d.py

```diff
@@ -117,10 +115,12 @@ def evaluate_t(self, x):
         nimgs = x.shape[2]
 
         # finufftpy require it to be aligned in fortran order
-        half_size = self.nrad * self.ntheta // 2
-        pf = np.empty((half_size, nimgs), dtype='complex128', order='F')
-        finufftpy.nufft2d2many(self.freqs[0, :half_size], self.freqs[1, :half_size], pf, 1, 1e-15, x)
-        pf = m_reshape(pf, (self.nrad, self.ntheta // 2, nimgs))
+        half_size = self.ntheta // 2
+        pf = np.empty((self.nrad * half_size, nimgs), dtype='complex128', order='F')
```

Comment:
Shouldn't the precision match that of `x` here? Also is F-order needed?

Reply:
The `x` is a real value but `pf` is a complex number. The F-order is removed.

Reply:
Yes but if `x` is `float32`, shouldn't `pf` be `complex64`?

Reply:
OK. Would you like to make the float32 or float64?

Reply:
Make what float32 or float64?

Reply:
Actually in the base class, there is a self.dtype witch a default value of float64. I believe Garrett added recently. Should I use it in default?

Reply:
Ok. I'm guessing that dtype is there to make sure that all calculations with this basis object happens with that dtype. (@garrettwrong correct me here if that's not what you intended.) In that case, we should use that dtype here but also make sure that the input (for both `evaluate` and `evaluate_t`) has the right dtype.

Reply:
I can't look easily on phone but if it is the line i think, then yes, the class has dtype which we should try to respect internally.  To be honest, I didn't even notice. Easy to do. I have issue for existing code, but it is nice if improve when tinkering if it isn't a distraction.  Complex is as you suggest. I found we were initializing certain classes with one type and then some things inside with different type, then operating with ? and this was making mixed modes.  You find those things when you start passing void pointers 🙃. Sorry I can't be more help today.

Reply:
Ok then let's do that. Check that input is consistent in terms of `self.dtype`, that is, for `self.dtype == 'float32'`, we should have `x.dtype == 'float32'` and `v.dtype == 'complex64'` and for `self.dtype == 'float64'`, we should have `x.dtype == 'float64'` and `v.dtype == 'complex128'`.

Reply:
Fixed it in commit  df7e893.

Reply:
We still need to check that the input dtype is consistent with `self.dtype`.

Reply:
If the input is not consistent with self.dtype, which one would you like to use?

Reply:
That is an excellent question. Some codes error... If we change/cast it, could we at least log it?

Reply:
I think we decided on erroring when dtypes mismatch. Seems like the safest thing to do (but shouldn't come up that often).